radeon-crimson: download failed. curl: option -: is unknown

## Issue description

```
building '/nix/store/w5nxdk461d9z3245qw77a7arjvy57s98-radeon-crimson-15.12-15.302-151217a-297685e.zip.drv'...
building '/nix/store/9h53aw49mhhymh4hmmhhzrz9nv76qz63-xorg-server-1.17.4.drv'...
building '/nix/store/38cyl426hfvzrc5il3bhxd5nxrjj2v7m-nixos-manual-combined.drv'...
building '/nix/store/wxjdipjn9z2k3p83hjsh6d8lz7cqn272-xwayland-1.17.4.drv'...
waiting for locks or build slots...
error checking the existence of http://tarballs.nixos.org/sha256/0n0ynqmjkjp5dl5q07as7ps3rlyyn63hq4mlwgd7c7v82ky2skvh:
curl: (7) Failed to connect to 0 port 80: Connection refused
curl: (22) The requested URL returned error: 404 Not Found

trying https://www2.ati.com/drivers/linux/radeon-crimson-15.12-15.302-151217a-297685e.zip
curl: option -: is unknown
curl: try 'curl --help' for more information
error: cannot download radeon-crimson-15.12-15.302-151217a-297685e.zip from any mirror
builder for '/nix/store/w5nxdk461d9z3245qw77a7arjvy57s98-radeon-crimson-15.12-15.302-151217a-297685e.zip.drv' failed with exit code 1
cannot build derivation '/nix/store/yka3lsw47il6ykvdik06hd6sw5y70s90-ati-drivers-15.12-4.14.78.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lazl5jgzz3d74ckikd1mw1i6asfyj06k-etc.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/4w2kn7x2rwdy3m2n8an227fxqwz0g2fl-kernel-modules.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lcf8fgz18j7k3d5gd201kx2wqbd0avis-opengl-drivers.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/f6rkslb8mxk7v44z1bcgizmyd4l12axi-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/30wgfc83ih8n4x9kmhakrkp4ygbrrflp-nixos-system-imac-18.09.1023.06fb0253afa.drv': 1 dependencies couldn't be built
error: build of '/nix/store/30wgfc83ih8n4x9kmhakrkp4ygbrrflp-nixos-system-imac-18.09.1023.06fb0253afa.drv' failed
```

Maybe it's a problem with the `curlOpts`? https://github.com/NixOS/nixpkgs/blob/2c2f1e37d4374ea61caefd9389927ea03df4ce31/pkgs/os-specific/linux/ati-drivers/default.nix#L65

Don't seem so. I can also not download it manually with curl:
```
curl -O --referer http://support.amd.com/en-us/download/desktop?os=Linux+x86_64 https://www2.ati.com/drivers/linux/radeon-crimson-15.12-15.302-151217a-297685e.zip
```
(i get a 0 byte file)

But i can download it using a browser from the website. But the hash is also different:

```
[root@imac:~]# sha256sum radeon-crimson-15.12-15.302-151217a-297685e.zip 
704f2dfc14681f76dae3b4120c87b1ded33cf43d5a1d800b6de5ca292bb61e58  radeon-crimson-15.12-15.302-151217a-297685e.zip
```

So they prevent us from downloading the file using something else than their website. Maybe the referer is not enough now?

### Steps to reproduce

1.
```
  services.xserver.videoDrivers = [ "ati_unfree" ];
  hardware.opengl.driSupport32Bit = true;
  nixpkgs.config.allowUnfree = true;
```
2. `nixos-rebuild switch`

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.78, NixOS, 18.09.1023.06fb0253afa (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.1`
 - channels(root): `"nixos-18.09.1023.06fb0253afa, nixos-hardware, nixos-unstable-19.03pre156996.c70ad805d21"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

I can fix the download error with this patch, but build fails now.

``` diff
diff --git a/pkgs/os-specific/linux/ati-drivers/default.nix b/pkgs/os-specific/linux/ati-drivers/default.nix
index d5d42faa913..cda7e03cd06 100644
--- a/pkgs/os-specific/linux/ati-drivers/default.nix
+++ b/pkgs/os-specific/linux/ati-drivers/default.nix
@@ -61,8 +61,8 @@ stdenv.mkDerivation rec {
   src = fetchurl {
     url =
     "https://www2.ati.com/drivers/linux/radeon-crimson-15.12-15.302-151217a-297685e.zip";
-    sha256 = "0n0ynqmjkjp5dl5q07as7ps3rlyyn63hq4mlwgd7c7v82ky2skvh";
-    curlOpts = "--referer http://support.amd.com/en-us/download/desktop?os=Linux+x86_64";
+    sha256 = "704f2dfc14681f76dae3b4120c87b1ded33cf43d5a1d800b6de5ca292bb61e58";
+    curlOpts = "--referer https://www.amd.com/en/support";
   };
 
   hardeningDisable = [ "pic" "format" ];

```

Maybe they have changed the content of the file, so also the hash has changed.

```
checking if gcc supports -Wformat=2... yes
+ for pelib1 in fglrx_dri.so dri/fglrx_dri.so
+ patchelf --remove-needed libX11.so.6 /nix/store/bzvxw3xn7061ny9w5z13cdb6nvpcwl97-ati-drivers-15.12/lib/fglrx_dri.so
checking if gcc supports -Wdeclaration-after-statement... yes
checking if gcc supports -Wstrict-prototypes... yes
+ for pelib1 in fglrx_dri.so dri/fglrx_dri.so
+ patchelf --remove-needed libX11.so.6 /nix/store/bzvxw3xn7061ny9w5z13cdb6nvpcwl97-ati-drivers-15.12/lib/dri/fglrx_dri.so
checking if gcc supports -Wunused... yes
+ for pelib2 in libatiadlxx.so xorg/modules/glesx.so dri/fglrx_dri.so fglrx_dri.so libaticaldd.so
+ patchelf --set-rpath /nix/store/a0d6q18gxggcvb5x24nrdg6nvhbnipnb-glibc-2.27/lib/:/nix/store/jzlj3z1z8wf0qjdjd4460br66xkr17gm-gcc-7.3.0-lib/lib/ /nix/store/bzvxw3xn7061ny9w5z13cdb6nvpcwl97-ati-drivers-15.12/lib/libatiadlxx.so
+ for pelib2 in libatiadlxx.so xorg/modules/glesx.so dri/fglrx_dri.so fglrx_dri.so libaticaldd.so
+ patchelf --set-rpath /nix/store/a0d6q18gxggcvb5x24nrdg6nvhbnipnb-glibc-2.27/lib/:/nix/store/jzlj3z1z8wf0qjdjd4460br66xkr17gm-gcc-7.3.0-lib/lib/ /nix/store/bzvxw3xn7061ny9w5z13cdb6nvpcwl97-ati-drivers-15.12/lib/xorg/modules/glesx.so
+ for pelib2 in libatiadlxx.so xorg/modules/glesx.so dri/fglrx_dri.so fglrx_dri.so libaticaldd.so
+ patchelf --set-rpath /nix/store/a0d6q18gxggcvb5x24nrdg6nvhbnipnb-glibc-2.27/lib/:/nix/store/jzlj3z1z8wf0qjdjd4460br66xkr17gm-gcc-7.3.0-lib/lib/ /nix/store/bzvxw3xn7061ny9w5z13cdb6nvpcwl97-ati-drivers-15.12/lib/dri/fglrx_dri.so
checking if gcc supports -Wmissing-prototypes... yes
checking if gcc supports -Wuninitialized... yes
/nix/store/13r1dybm9f1xz0bl2s33qqi36cypbxzd-linux-4.14.78-dev/lib/modules/4.14.78/source/Makefile:953: *** "Cannot generate ORC metadata for CONFIG_UNWINDER_ORC=y, please install libelf-dev, libelf-devel or elfutils-libelf-devel".  Stop.
make[2]: *** [Makefile:146: sub-make] Error 2
make[1]: *** [Makefile:24: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/13r1dybm9f1xz0bl2s33qqi36cypbxzd-linux-4.14.78-dev/lib/modules/4.14.78/build'
make: *** [Makefile:88: kmod_build] Error 2
+ exitHandler
+ exitCode=2
+ set +e
+ '[' -n '' ']'
+ ((  2 != 0  ))
+ runHook failureHook
++ shopt -po nounset
+ local 'oldOpts=set +o nounset'
+ set -u
+ local hookName=failureHook
+ shift
+ local 'hooksSlice=failureHooks[@]'
+ local hook
+ for hook in "_callImplicitHook 0 $hookName" ${!hooksSlice+"${!hooksSlice}"}
+ _eval '_callImplicitHook 0 failureHook'
++ type -t '_callImplicitHook 0 failureHook'
+ '[' '' = function ']'
+ set +u
+ eval '_callImplicitHook 0 failureHook'
++ _callImplicitHook 0 failureHook
++ set -u
++ local def=0
++ local hookName=failureHook
++ case "$(type -t "$hookName")" in
+++ type -t failureHook
++ '[' -z '' ']'
++ return 0
+ set -u
+ eval 'set +o nounset'
++ set +o nounset
+ return 0
+ '[' -n '' ']'
+ exit 2
builder for '/nix/store/z2rhgqkfvg6brw2lq1fsblvj31x1s0qr-ati-drivers-15.12-4.14.78.drv' failed with exit code 2
```

I would like to also fix the build before creating a PR, but i have no idea how to fix this.

--- a/pkgs/os-specific/linux/ati-drivers/default.nix
+++ b/pkgs/os-specific/linux/ati-drivers/default.nix
@@ -61,8 +61,8 @@ stdenv.mkDerivation rec {
   src = fetchurl {
     url =
     "https://www2.ati.com/drivers/linux/radeon-crimson-15.12-15.302-151217a-297685e.zip";
-    sha256 = "0n0ynqmjkjp5dl5q07as7ps3rlyyn63hq4mlwgd7c7v82ky2skvh";
-    curlOpts = "--referer http://support.amd.com/en-us/download/desktop?os=Linux+x86_64";
+    sha256 = "704f2dfc14681f76dae3b4120c87b1ded33cf43d5a1d800b6de5ca292bb61e58";
+    curlOpts = "--referer https://www.amd.com/en/support";
   };
 
   hardeningDisable = [ "pic" "format" ];

